src/js/core.js

```diff
@@ -0,0 +1,126 @@
+var d3 = require('d3');
+var _ = require('underscore');
+var utils = require('./utils');
+
+function compute_svg_width(rect_width, rect_padding, row_length) {
+  return (rect_width + rect_padding) * row_length;
+}
+
+var core = function() {
+  var config = { row_height: 15 };
+  var container_width = 100;
+  var element_padding = 1;
+  var element_width = 1;
+  var labels = [];
+  var rows = undefined;
+  var svg_height = 95;
+
+  // styles, appends, does all the right stuff to the container
+  // so that we can go on to work with the inner <svg>.
+  function oncoprint_container_to_svg(container) {
+    container.style('width', container_width + "px")
+    .style('display', 'inline-block')
+    .style('overflow-x', 'auto')
+    .style('overflow-y', 'hidden');
+
+    return container.append('svg')
+    .attr('width', compute_svg_width(element_width, element_padding, rows[0].length))
+    .attr('height', config.row_height * rows.length);
+  }
+
+  var me = function(container) {
+
```

Comment:
Do you mean in the `oncoprint_container_to_svg` function? Or do you mean in the `me` function? The core module is still under construction mainly because I haven't worked out how to instruct core which rendering function to use for which row. Once I figure that out, a bunch of stuff will get deleted here.

I'm going between asking the user to directly specify, for each row, which rendering function to use and then binding them with something like:

```js
oncoprint.setRenderingRules([function(selection) {...}, function(selection) { ... }])
```

And by setting up some sort of system where the user provided a function which decided which rendering function to use based on the type of data it was given.

Now that I write this I see that the first solution is far simpler and I think I'll just go with that. Yes, the user will have to be a bit careful, but not big deal.